comtypes/test/test_midl_safearray_create.py

```diff
@@ -0,0 +1,93 @@
+# coding: utf-8
+
+from ctypes import byref, c_int, pointer, POINTER
+import unittest
+
+import comtypes
+import comtypes.safearray
+from comtypes import CLSCTX_INPROC_SERVER, CLSCTX_LOCAL_SERVER
+from comtypes.client import CreateObject, GetModule
+import comtypes.typeinfo
+
+GetModule("UIAutomationCore.dll")
+from comtypes.gen.UIAutomationClient import CUIAutomation, IUIAutomation
+
+ComtypesCppTestSrvLib_GUID = "{07D2AEE5-1DF8-4D2C-953A-554ADFD25F99}"
+
+try:
+    GetModule((ComtypesCppTestSrvLib_GUID, 1, 0, 0))
+    from comtypes.gen.ComtypesCppTestSrvLib import StructRecordParamTest
+    from comtypes.gen.ComtypesCppTestSrvLib import IDispSafearrayParamTest
+
+    IMPORT_FAILED = False
+except (ImportError, OSError):
+    IMPORT_FAILED = True
+
+
+class Test_midlSAFEARRAY_create(unittest.TestCase):
+    def test_iunk(self):
+        extra = pointer(IUIAutomation._iid_)
+        iuia = CreateObject(
+            CUIAutomation().IPersist_GetClassID(),
+            interface=IUIAutomation,
+            clsctx=CLSCTX_INPROC_SERVER,
+        )
+        sa_type = comtypes.safearray._midlSAFEARRAY(POINTER(IUIAutomation))
+        for ptn, sa in [
+            ("with extra", sa_type.create([iuia], extra=extra)),
+            ("without extra", sa_type.create([iuia])),
+        ]:
+            with self.subTest(ptn=ptn):
+                (unpacked,) = sa.unpack()
+                self.assertIsInstance(unpacked, POINTER(IUIAutomation))
+
+    @unittest.skipIf(IMPORT_FAILED, "This depends on the out of process COM-server.")
+    def test_idisp(self):
+        extra = pointer(IDispSafearrayParamTest._iid_)
+        idisp = CreateObject(
+            "Comtypes.DispSafearrayParamTest",
+            clsctx=CLSCTX_LOCAL_SERVER,
+            interface=IDispSafearrayParamTest,
+        )
+        sa_type = comtypes.safearray._midlSAFEARRAY(POINTER(IDispSafearrayParamTest))
+        for ptn, sa in [
+            ("with extra", sa_type.create([idisp], extra=extra)),
+            ("without extra", sa_type.create([idisp])),
+        ]:
+            with self.subTest(ptn=ptn):
+                (unpacked,) = sa.unpack()
+                self.assertIsInstance(unpacked, POINTER(IDispSafearrayParamTest))
+
+    @unittest.skipIf(IMPORT_FAILED, "This depends on the out of process COM-server.")
+    def test_record(self):
+        extra = comtypes.typeinfo.GetRecordInfoFromGuids(
+            *StructRecordParamTest._recordinfo_
+        )
+        record = StructRecordParamTest()
+        record.answer = 42
```

Comment:
I think we should also set some values for `question` and `needs_clarification`, and assert that `unpacked` holds the same values.